bookmarks/vanilla/src/modules/Bookmark.ts

```diff
@@ -122,31 +130,31 @@ export class Bookmark {
      * @param title string
      * @param link string
      * @param session Session
-     * @returns IBookmark[]
+     * @returns IBookmark
      */
-    public static async create(title: string, link: string, session: Session) {
+    public static async create(payload: ICreateBookmark, session: Session) {
+
+        const { title, link, creator, topic } = payload
 
         const indexUrl = await this.getIndexUrl(session);
 
         const ds = await getSolidDataset(indexUrl, { fetch: session.fetch });
 
-        const newBookmarkThing = buildThing(createThing())
-            .addStringNoLocale(DCTERMS.title, title)
-            .addStringNoLocale(BOOKMARK.recalls, link)
-            .addUrl(RDF.type, BOOKMARK.Bookmark)
-            .build();
+        let newBookmarkThing = createThing()
+
+        newBookmarkThing = addStringNoLocale(newBookmarkThing, DCTERMS.title, title)
+        newBookmarkThing = addStringNoLocale(newBookmarkThing, BOOKMARK.recalls, link)
+        if (creator) newBookmarkThing = addNamedNode(newBookmarkThing, DCTERMS.creator, namedNode(creator))
+        if (topic) newBookmarkThing = addNamedNode(newBookmarkThing, BOOKMARK.hasTopic, namedNode(topic))
+        newBookmarkThing = addStringNoLocale(newBookmarkThing, DCTERMS.created, new Date().toISOString())
+        newBookmarkThing = addStringNoLocale(newBookmarkThing, "http://purl.org/dc/terms/updated", new Date().toISOString())
+
+        newBookmarkThing = addUrl(newBookmarkThing, RDF.type, BOOKMARK.Bookmark)
 
         const updatedBookmarkList = setThing(ds, newBookmarkThing);
-        const updatedDataset = await saveSolidDatasetAt(indexUrl, updatedBookmarkList, { fetch: session.fetch });
-        const things = getThingAll(updatedDataset)
-
-        return things.map(thing => {
-            return {
-                url: thing.url,
-                title: getLiteral(thing, DCTERMS.title)?.value,
-                link: getLiteral(thing, BOOKMARK.recalls)?.value
-            }
-        }) as IBookmark[]
+        await saveSolidDatasetAt(indexUrl, updatedBookmarkList, { fetch: session.fetch });
+        // TODO: also need return url of created bookmark
```

Comment:
I think create is incomplete unless we also return the URL of the created bookmark.

Reply:
it's not possible yet, at least for now
returning a boolean there.